### PR TITLE
fix monster attack behavior

### PR DESCRIPTION
**Pull request by ArturKnopik**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Fix monster attack behavior

**Issues addressed:** 
https://github.com/otland/forgottenserver/issues/4865

**How to test:** 
go to monsters using normal character
Troll - hit to red, let them fleeing
Cyclops
Hunter 
Rabbit 
IDK if there is monster with different walking mechanic

### CODE REVIEW

**Comment by nekiro** [comment]
imo there is missing updateFollowCreaturePath(fpp); after this code so specifically:

https://github.com/otland/forgottenserver/blob/a78d2ed78fab73069d429e18d4929f77aaf9db5c/src/monster.cpp#L606
```
} else {
    updateFollowCreaturePath(fpp);
}
```
genuinely sad this code was merged, its broken at the first sight when **joining the game and checking**

**Comment by ramon-bernardo** [comment]
@nekiro I don't see where this `update` might be missing, as redundancies were removed (**highlighted in red**), and the logic was simplified (**highlighted in yellow**)

**Comment by ArturKnopik** [comment]
@nekiro @ramon-bernardo, can you test my fix?

File: src/monster.cpp [hunk 1]
```diff
@@ -605,10 +612,14 @@ void Monster::goToFollowCreature()
 			startAutoWalk();
 		}
 	} else {
-		updateFollowCreaturePath(fpp);
```
Comment (nekiro) [inline src/monster.cpp:608]: use updateFollowCreaturePath
its the exact duplicate of the code you pasted here, so there is no reason to duplicate the code again

File: src/monster.cpp [hunk 2]
```diff
@@ -578,21 +578,28 @@ bool Monster::searchTarget(TargetSearchType_t searchType /*= TARGETSEARCH_DEFAUL
...
+	if (getMaster() && (isFleeing() || fpp.maxTargetDist > 1)) {
 		Direction dir = DIRECTION_NONE;
 
 		if (isFleeing()) {
 			getDistanceStep(followCreature->getPosition(), dir, true);
-		} else if (fpp.maxTargetDist > 1) {
+		} else { // maxTargetDist > 1
 			if (!getDistanceStep(followCreature->getPosition(), dir)) {
 				// if we can't get anything then let the A* calculate
-				updateFollowCreaturePath(fpp);
```
Comment (nekiro) [inline src/monster.cpp:595]: same as previously, use this method as its the same code, so its okay

**Review by nekiro — CHANGES_REQUESTED** [review]

**Comment by ramon-bernardo** [comment]
Unfortunately, i cant test it because my cip client doesn't connect to login server. I could try using OTClient, but not sure if it support v13

**Review by nekiro — APPROVED** [review]

**Pull request merged**